src/meta-client/src/client/ddl.rs

```diff
@@ -133,23 +135,41 @@ impl Inner {
         loop {
             if let Some(leader) = &ask_leader.get_leader() {
                 let mut client = self.make_client(leader)?;
-                let res = client
+
+                match client
                     .submit_ddl_task(req.clone())
                     .await
-                    .context(error::TonicStatusSnafu)?;
-
-                let res = res.into_inner();
-
-                if let Some(header) = res.header.as_ref() {
-                    if let Some(err) = header.error.as_ref() {
-                        if err.code == ErrorCode::NotLeader as i32 {
-                            let _ = ask_leader.ask_leader().await?;
+                    .context(error::TonicStatusSnafu)
```

Comment:
Lack of tests. I think it's a good opportunity to introduce "failpoints" to our codes.

Reply:
It's taken longer than expected; let's leave this to future works.